service/src/main/java/com/codeforcommunity/database/tableImpl/PostTable.java

```diff
@@ -0,0 +1,93 @@
+  // Our in-memory database.
+  private final Map<Integer, SinglePostResponse> postMap;
+  // A count of how many posts we want to make on startup for the in memory db. You can change this
+  // to be whatever you want.
+  public static final int STUB_POST_COUNT = 15;
+  // The classic Lorem Ipsum text. For now this will be our stubbed data.
+  public static final String LOREM_IPSUM_TEXT =
+      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt "
+          + "ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation "
+          + "ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in "
+          + "reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. "
+          + "Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia "
+          + "deserunt mollit anim id est laborum.";
+
+  /** The constructor we'll be using in production. */
+  public PostTable() {
+    this.postMap = new HashMap<>();
+  }
+
+  /**
+   * The constructor we'll be using when we want the db to be seeded with fake data.
+   *
+   * @param postCount The count of posts to create
+   */
+  public PostTable(int postCount) {
+    // Call the other constructor to set up the map.
+    this();
+
+    // Initialize a new Random object.
+    Random random = new Random();
+
+    // Loop through and create stubPostCount amount of posts.
+    for (int i = 0; i < postCount; i++) {
+      // Build a date string. For each number to be inserted, a random number from [0, bound) is
+      // selected, and then is transformed to look more like a date. 1 is added to the day and month
+      // so that you'll have days between [1, 30], and months between [1, 12]. 2000 is added to the
+      // year so you'll get years between 2000 and 2020.
+      String date =
+          String.format(
+              "%d/%d/%d",
+              random.nextInt(12) + 1, random.nextInt(30) + 1, random.nextInt(21) + 2000);
+      // Get a length we'll want to use to select text.
+      int textLength = random.nextInt(LOREM_IPSUM_TEXT.length());
+      // Select a substring of textLength characters from the Lorem Ipsum text.
+      String body = LOREM_IPSUM_TEXT.substring(0, textLength);
+      // Create a new post with the generated data from above.
+      SinglePostResponse post =
+          new SinglePostResponse(
+              i, "Author " + i % 5, date, "Post #" + i, random.nextInt(1000), body);
+      this.savePost(post);
+    }
+  }
+
+  @Override
+  public SinglePostResponse getById(int id) {
+    return this.postMap.get(id);
+  }
+
+  @Override
+  public List<SinglePostResponse> getAllPosts() {
+    // Get the values from the postMap, convert them to a Post[] (by passing in a new Post[] with no
+    // space allocated), and then turn it into a List.
+    List<SinglePostResponse> posts =
+        Arrays.asList(this.postMap.values().toArray(new SinglePostResponse[0]));
```

Comment:
I'm pretty sure you can just do this as `new ArrayList<>(this.postMap.values());`

Reply:
Oh wait yeah I have no idea why I did that.